src/test/java/htsjdk/variant/variantcontext/VariantContextUnitTest.java

```diff
@@ -1651,4 +1651,21 @@ public void testGetAttributeAsStringList() {
         // test the case of a missing key
         Assert.assertTrue(context.getAttributeAsStringList("MissingList", "empty").isEmpty());
     }
+
+
+    @DataProvider
+    public Object[][] getVariantsWithID(){
+        final VariantContextBuilder vcb = new VariantContextBuilder("test", "chr1", 10L, 10L, Arrays.asList(Allele.REF_A, Allele.ALT_C));
+        return new Object[][]{
+                {vcb.noID().make(), Collections.<String>emptyList()},
+                {vcb.id(".").make(), Collections.<String>emptyList()},
+                {vcb.id("rs128").make(), Collections.singletonList("rs128")},
+                {vcb.id("1;2;rs128;soManyIDs").make(), Arrays.asList("1", "2", "rs128", "soManyIDs")}
```

Comment:
What should happen if one of the ids is '.'?

Reply:
or if one of the ids is empty (i.e. `idONE,,idThree`)

Reply:
I would argue that the description in vcfSpec isn't tight enough. it just says 
>   identifier: Semi-colon separated list of unique identifiers where available. If this is a dbSNP variant the
rs number(s) should be used. No identifier should be present in more than one data record. If there is no
identifier available, then the MISSING value should be used. (String, no white-space or semi-colons permitted,
duplicate values not allowed.)"

so it doesn't specifically mention that the single period is not allowed but it is the MISSING value....so IDK